### PR TITLE
Add doc for optimizer.plan-with-table-node-partitioning

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 1]
```diff
@@ -116,3 +116,14 @@ the maximum number of joins that can be reordered at once.
...
 Reduces number of rows produced by joins when optimizer detects that duplicated
 join output rows can be skipped.
+
+``optimizer.plan-with-table-node-partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+Use connector provided table node partitioning when reading tables.
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:126]: What is "connector provided table node partitioning" ?
Maybe a concrete example of a connector would help

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 2]
```diff
@@ -116,3 +116,14 @@ the maximum number of joins that can be reordered at once.
...
+When set to ``true`` each table partition is read by a separate node.
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:127]: When should a user consider setting it to false ?

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 3]
```diff
@@ -116,3 +116,18 @@ the maximum number of joins that can be reordered at once.
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+Use connector provided table node partitioning when reading tables.
+For example, table node partitioning corresponds to Hive table buckets.
+When set to ``true`` each table partition is read by a separate node.
+Partition reader assignments are distributed across worker nodes for
+parallel processing. Using of table scan node partitioning can improve
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:130]: Using of -> Use of

**Review by raunaqmorarka — APPROVED** [review]

**Review by raunaqmorarka — APPROVED** [review]

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 4]
```diff
@@ -116,3 +116,18 @@ the maximum number of joins that can be reordered at once.
...
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+Use connector provided table node partitioning when reading tables.
+For example, table node partitioning corresponds to Hive table buckets.
+When set to ``true`` each table partition is read by a separate worker.
+Partition reader assignments are distributed across workers for
+parallel processing. Use of table scan node partitioning can improve
+query performance by reducing query complexity (e.g cluster wide data reshuffle
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:131]: complexity. For example, cluster wide data reshuffling might not be needed when processing an aggregation query. However, ..

**Review by mosabua — CHANGES_REQUESTED** [review]
Update to avoid using e.g. and () ..

**Review by mosabua — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**